subprojects/ide/src/main/java/org/gradle/plugins/ide/internal/tooling/RunBuildDependenciesTaskBuilder.java

```diff
@@ -0,0 +1,150 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.plugins.ide.internal.tooling;
+
+import org.gradle.StartParameter;
+import org.gradle.api.Action;
+import org.gradle.api.NonNullApi;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.artifacts.ArtifactView;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.artifacts.component.ComponentIdentifier;
+import org.gradle.api.attributes.AttributeContainer;
+import org.gradle.api.attributes.Usage;
+import org.gradle.api.internal.artifacts.DefaultProjectComponentIdentifier;
+import org.gradle.api.internal.model.NamedObjectInstantiator;
+import org.gradle.api.internal.project.ProjectInternal;
+import org.gradle.api.internal.project.ProjectState;
+import org.gradle.api.internal.project.ProjectStateRegistry;
+import org.gradle.api.specs.Spec;
+import org.gradle.plugins.ide.eclipse.EclipsePlugin;
+import org.gradle.plugins.ide.eclipse.model.EclipseModel;
+import org.gradle.tooling.model.eclipse.EclipseRuntime;
+import org.gradle.tooling.provider.model.ParameterizedToolingModelBuilder;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@NonNullApi
+public class RunBuildDependenciesTaskBuilder implements ParameterizedToolingModelBuilder<EclipseRuntime> {
+    private final ProjectStateRegistry projectRegistry;
+
+    public RunBuildDependenciesTaskBuilder(ProjectStateRegistry projectRegistry) {
+        this.projectRegistry = projectRegistry;
+    }
+
+    @Override
+    public Class<EclipseRuntime> getParameterType() {
+        return EclipseRuntime.class;
+    }
+
+    @Override
+    public Object buildAll(String modelName, EclipseRuntime eclipseRuntime, Project project) {
+        Set<String> closedProjectPaths = gatherClosedProjectPaths(eclipseRuntime);
+        Set<? extends Task> buildDependencies = gatherBuildDependenciesForClosedProjects(closedProjectPaths);
+
+        if (!buildDependencies.isEmpty()) {
+            Project rootProject = project.getRootProject();
+            StartParameter startParameter = project.getGradle().getStartParameter();
+            List<String> taskPaths = new ArrayList<>(startParameter.getTaskNames());
+            String parentTaskName = parentTaskName(rootProject, "eclipseClosedDependencies");
+            Task task = rootProject.task(parentTaskName);
+            task.dependsOn(buildDependencies.toArray(new Object[0]));
+            taskPaths.add(parentTaskName);
+            startParameter.setTaskNames(taskPaths);
+        }
+        return new Object();
+    }
+
+    private Set<Task> gatherBuildDependenciesForClosedProjects(Set<String> closedProjectPaths) {
```

Comment:
❌ Same as in the EclipseModelBuilder, add the necessary information ("buildDependencies") to the `ProjectDependency` in the Eclipse dependency resolver instead of reimplementing resolution here.